### PR TITLE
PHOENIX-6430 Added support for full row update for tables when no columns specfied in scenario

### PR DESCRIPTION
**Pull request by jpisaac**

### CODE REVIEW

**Review by ChinmaySKulkarni — APPROVED** [review]
Reviewed the second commit only (PHOENIX-6430) and it lgtm.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   7m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 14 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 47s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 36s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 18s |  4.x passed  |
| +0 :ok: |  spotbugs  |   0m 58s |  phoenix-pherf in 4.x has 50 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 36s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 28s |  phoenix-pherf: The patch generated 123 new + 548 unchanged - 49 fixed = 671 total (was 597)  |
| -1 :x: |  whitespace  |   0m  1s |  The patch 8 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  4s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 17s |  the patch passed  |
| -1 :x: |  spotbugs  |   1m  8s |  phoenix-pherf generated 2 new + 50 unchanged - 0 fixed = 52 total (was 50)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 54s |  phoenix-pherf in the patch failed.  |
| -1 :x: |  asflicense  |   0m 10s |  The patch generated 1 ASF License warnings.  |
|  |   |  42m 41s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:phoenix-pherf |
|  |  Switch statement found in org.apache.phoenix.pherf.rules.RulesApplier.getRuleBasedDataGeneratorForColumn(Column) where default case is missing  At RulesApplier.java:where default case is missing  At RulesApplier.java:[lines 554-572] |
|  |  A prepared statement is generated from a nonconstant String in org.apache.phoenix.pherf.workload.mt.tenantoperation.PreScenarioOperationSupplier$1.apply(TenantOperationInfo)  At PreScenarioOperationSupplier.java:from a nonconstant String in org.apache.phoenix.pherf.workload.mt.tenantoperation.PreScenarioOperationSupplier$1.apply(TenantOperationInfo)  At PreScenarioOperationSupplier.java:[line 69] |
| Failed junit tests | phoenix.pherf.DataIngestIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1186 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux d5b04fbbe6f1 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3535708 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/whitespace-tabs.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/new-spotbugs-phoenix-pherf.html |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/patch-unit-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1793 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 19 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +0 :ok: |  mvndep  |   5m  8s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m 16s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m 38s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 53s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   1m  1s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m  2s |  phoenix-core in 4.x has 941 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 54s |  phoenix-pherf in 4.x has 50 extant spotbugs warnings.  |
| -0 :warning: |  patch  |   1m  8s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 37s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 29s |  phoenix-pherf: The patch generated 134 new + 543 unchanged - 54 fixed = 677 total (was 597)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 8 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  7s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m  1s |  the patch passed  |
| -1 :x: |  spotbugs  |   1m  7s |  phoenix-pherf generated 1 new + 50 unchanged - 0 fixed = 51 total (was 50)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 131m 40s |  phoenix-core in the patch passed.  |
| -1 :x: |  unit  |   0m 59s |  phoenix-pherf in the patch failed.  |
| -1 :x: |  asflicense  |   0m 19s |  The patch generated 1 ASF License warnings.  |
|  |   | 170m 45s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:phoenix-pherf |
|  |  A prepared statement is generated from a nonconstant String in org.apache.phoenix.pherf.workload.mt.tenantoperation.PreScenarioOperationSupplier$1.apply(TenantOperationInfo)  At PreScenarioOperationSupplier.java:from a nonconstant String in org.apache.phoenix.pherf.workload.mt.tenantoperation.PreScenarioOperationSupplier$1.apply(TenantOperationInfo)  At PreScenarioOperationSupplier.java:[line 76] |
| Failed junit tests | phoenix.pherf.workload.mt.tenantoperation.TenantOperationEventGeneratorTest |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1186 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 7b7467530f90 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3535708 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/whitespace-tabs.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/new-spotbugs-phoenix-pherf.html |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/patch-unit-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 5950 (vs. ulimit of 30000) |
| modules | C: phoenix-core phoenix-pherf U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-pherf/src/it/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/MultiTenantTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
+package org.apache.phoenix.pherf.workload.mt.tenantoperation;
```
Comment (yanxinyi) [inline phoenix-pherf/src/it/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/MultiTenantTestUtils.java:1]: apache header

File: phoenix-pherf/src/main/java/org/apache/phoenix/pherf/rules/RulesApplier.java [hunk 2]
```diff
@@ -35,14 +35,12 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.text.SimpleDateFormat;
 import java.util.*;
```
Comment (yanxinyi) [inline phoenix-pherf/src/main/java/org/apache/phoenix/pherf/rules/RulesApplier.java:38]: nit: maybe remove import *?

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 52s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  1s |  The patch appears to include 9 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  16m 16s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 45s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 21s |  4.x passed  |
| +0 :ok: |  spotbugs  |   1m 12s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 44s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 17s |  phoenix-pherf: The patch generated 26 new + 176 unchanged - 17 fixed = 202 total (was 193)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  4s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 17s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  11m 37s |  phoenix-pherf in the patch passed.  |
| -1 :x: |  asflicense  |   0m 10s |  The patch generated 1 ASF License warnings.  |
|  |   |  43m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1186 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux 7b34d855856f 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3f8a74e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1762 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 46s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 52s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 46s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 21s |  4.x passed  |
| +0 :ok: |  spotbugs  |   1m 14s |  phoenix-pherf in 4.x has 51 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 42s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 37s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 15s |  phoenix-pherf: The patch generated 26 new + 176 unchanged - 17 fixed = 202 total (was 193)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m  7s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  10m  1s |  phoenix-pherf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  39m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1186 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux a244da84b5ed 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3f8a74e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/4/artifact/yetus-general-check/output/diff-checkstyle-phoenix-pherf.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/4/testReport/ |
| Max. process+thread count | 2057 (vs. ulimit of 30000) |
| modules | C: phoenix-pherf U: phoenix-pherf |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1186/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by yanxinyi — APPROVED** [review]

**Pull request merged**